src/main.cpp

```diff
@@ -5446,15 +5546,152 @@ bool TallyResearchAverages()
     double NetworkPayments = 0;
     double NetworkInterest = 0;
     
+                        //Consensus Start/End block:
+                        int nMaxDepth = (nBestHeight-CONSENSUS_LOOKBACK) - ( (nBestHeight-CONSENSUS_LOOKBACK) % BLOCK_GRANULARITY);
+                        int nLookback = BLOCKS_PER_DAY * 14; //Daily block count * Lookback in days
+                        int nMinDepth = (nMaxDepth - nLookback) - ( (nMaxDepth-nLookback) % BLOCK_GRANULARITY);
+                        if (fDebug3) printf("START BLOCK %f, END BLOCK %f ",(double)nMaxDepth,(double)nMinDepth);
+                        if (nMinDepth < 2)              nMinDepth = 2;
+    if(fDebug) printf("TallyResearchAverages_retired: start %d end %d\n",nMaxDepth,nMinDepth);
+                        mvMagnitudesCopy.clear();
+                        int iRow = 0;
+                        //CBlock block;
+                        CBlockIndex* pblockindex = pindexBest;
+                        if (!pblockindex)
+                        {
+                                bTallyStarted_retired = false;
+                                bNetAveragesLoaded = true;
+                                return true;
+                        }
+                        while (pblockindex->nHeight > nMaxDepth)
+                        {
+                            if (!pblockindex || !pblockindex->pprev || pblockindex == pindexGenesisBlock) return false;
+                            pblockindex = pblockindex->pprev;
+                        }
+
+                        if (fDebug3) printf("Max block %f, seektime %f",(double)pblockindex->nHeight,(double)GetTimeMillis()-nStart);
```

Comment:
would love the unnecessary doubles removed. should be %d and PRId64 for those two.